File: app/src/main/java/com/yeputra/moviecatalogue/model/Movie.java
package com.yeputra.moviecatalogue.model;

import android.os.Parcel;
import android.os.Parcelable;

public class Movie implements Parcelable {
    private int icon;
    private String title;
    private String description;
    private String date;

    public Movie(int icon, String title, String description, String date) {
        this.icon = icon;
        this.title = title;
        this.description = description;
        this.date = date;
    }

    public int getIcon() {
        return icon;
    }

    public String getTitle() {
        return title;
    }

    public String getDescription() {
        return description;
    }

    public String getDate() {
        return date;
    }

    private Movie(Parcel in) {
        icon = in.readInt();
        title = in.readString();
        description = in.readString();
        date = in.readString();
    }

    public static final Creator<Movie> CREATOR = new Creator<Movie>() {
        @Override
        public Movie createFromParcel(Parcel in) {
            return new Movie(in);
        }

        @Override
        public Movie[] newArray(int size) {
            return new Movie[size];
        }
    };

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeInt(icon);
        dest.writeString(title);
        dest.writeString(description);
        dest.writeString(date);
    }
}

File: app/src/main/java/com/yeputra/moviecatalogue/model/Movie.kt
package com.yeputra.moviecatalogue.model

import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import kotlinx.android.parcel.Parcelize

@Parcelize
data class Movie(
        val id: Int?,
        @SerializedName("title", alternate = ["name"])
        val title: String?,
        val overview: String?,
        val release_date: String?,
        val vote_average: Double?,
        val backdrop_path: String?,
        val poster_path: String?,
        val adult: Boolean?,
        val genre_ids: MutableList<Int>?,
        val original_language: String?,
        @SerializedName("original_title", alternate = ["original_name"])
        val original_title: String?,
        val popularity: Double?,
        val video: Boolean?,
        val vote_count: Int?
) : Parcelable

@Parcelize
data class MovieFavorite (
        val id: String?,
        val title: String?,
        val origTitle: String?,
        val overview: String?,
        val adult: Boolean?,
        val release_date: String?,
        val vote_average: String?,
        val backdrop_path: String?,
        val poster_path: String?,
        val type: String?
) : Parcelable

enum class FilmType {
    MOVIE, TVSHOW
}